include/runtime/instance/memory.h

```diff
@@ -241,16 +270,11 @@ class MemoryInstance {
   /// \param Length the load length from data. Need to <= sizeof(T).
   ///
   /// \returns void when success, ErrCode when failed.
-  template <typename T>
+  template <typename T, uint32_t const Length = sizeof(T)>
   typename std::enable_if_t<IsWasmNumV<T>, Expect<void>>
-  loadValue(T &Value, uint32_t Offset, uint32_t Length) const noexcept {
+  loadValue(T &Value, uint32_t Offset) const noexcept {
     // Check the data boundary.
-    if (unlikely(Length > sizeof(T))) {
-      spdlog::error(ErrCode::MemoryOutOfBounds);
-      spdlog::error(
-          ErrInfo::InfoBoundary(Offset, Length, Offset + sizeof(T) - 1));
-      return Unexpect(ErrCode::MemoryOutOfBounds);
-    }
+    static_assert(Length <= sizeof(T));
```

Comment:
Please don't remove the boundary checking here.
For the memory load out of boundary here, we should return error code but not assert.

Reply:
Both the `Length` and `sizeof(T)` can be known at compile time and `static_assert` will do the check at compile time and fail the compile if the condition is not met.

Do we have any cases and any op that will explicitly load and store out of bound memory in our WasmEdge runtime code? If not, it will be beneficial to use `static_assert` here. First it can prevent us from writing incorrect code by accident. Second it avoids an unnecessary comparison at runtime and avoids a potential branch miss.

Reply:
Although the `Length` can be known at compile time, the instructions can only be known at runtime.
For WASM, we can only get the `Length` by different instructions at runtime and output the error message.

Reply:
Update:

Sorry that I misunderstood the code.
The boundary checking will be done in the following 2 lines.
You're right, this can be checked with `static_assert`. Thanks.